### DISCUSSION TITLE
Error in qm9_ecc.py

### DISCUSSION THREAD

**Discussion by behroozazarkhalili** [danielegrattarola/spektral#224] [Q&A]
I have received this error when running the following example with **BatchLoader**.
[https://github.com/danielegrattarola/spektral/blob/master/examples/graph_prediction/qm9_ecc.py](url)

`ValueError: Layer model_6 expects 4 input(s), but it received 3 input tensors. Inputs received: [<tf.Tensor 'inputs:0' shape=(None, None, 10) dtype=float64>, <tf.Tensor 'inputs_1:0' shape=(None, None, None) dtype=int64>, <tf.Tensor 'inputs_2:0' shape=(None, None, None, 4) dtype=float64>]`

Please mention the philosophy of the` I_in` layer in the code using **DisjointLoader**.

**Comment by danielegrattarola** [comment]
Hi, 

that example is outdated, refer to the new version currently in the `develop` branch: https://github.com/danielegrattarola/spektral/blob/develop/examples/graph_prediction/qm9_ecc.py

If you're using the BatchLoader you might want to check the version of that example that uses BatchLoader: https://github.com/danielegrattarola/spektral/blob/develop/examples/graph_prediction/qm9_ecc_batch.py

Also, the meaning of the `I` tensor is described here: https://graphneural.network/data-modes/#disjoint-mode
It's basically an integer map between nodes and the graph to which they belong.

Cheers
